expertisesolutions/efl
# Windows: Eina: Einafy unPOSIX sys/time.h

Windows: Eina: Einafy unPOSIX sys/time.h

Summary:
========
This PR reorganizes `unposix/sys/time.h` and as specified by issue #40,
this PR moves `unposix/sys/time.{h,c}` to `eina/eina_time.{h,c}` which
implements `eina_` wrap functions to `gettimeofday` and `localtime_r`.

In case of a Windows build it also defines struct `timezone` and `u_short`;
In case of a Linux build it includes `sys/time.h`;

Every occurrence of `gettimeofday` was moved to `eina_gettimeofday`;
Every occurrence of `localtime_r` was moved to `eina_localtime_r`;
Every `#include` of `sys/time.h` was removed and it was changed to
`#include "eina_time.h` in case of an in `eina` include or
`#include <Eina.h>` in case of an out of `eina` include;

`evil/evil_time.{c,h}` `sys/time.h` include was changed to `time.h` as it
only used `time_t`;
`evil/evil_unistd.c` again only uses `time_t`, and as `it includes
`evil_private.h` it will implicitly include `evil_time.h`;

Linux Test Plan:
================
- configure meson with `bindigns=cxx,mono`;
- `ninja` should return only errors/warnings that `master` does;
- `meson test all` should only pass on tests that master does;

Windows Test Plan:
==================
- run `configure.bat`;
- run `build.bat` with `-k0`;
- there should not be any errors/warnings about any related to `sys/time.h`;
- It should yield the same errors/warnings that `devs/expertise/native-windows`
  does;
